grai-server/app/connections/task_helpers.py

```diff
@@ -0,0 +1,85 @@
+from typing import Dict, List, Optional, TypeVar
+
+from grai_client.schemas.edge import Edge, EdgeV1
+from grai_client.schemas.node import Node, NodeV1, NodeID
+from grai_client.schemas.utilities import merge_models
+from lineage.models import Node as NodeModel, Edge as EdgeModel
+from workspaces.models import Workspace
+
+T = TypeVar("T", Node, Edge)
+
+
+def get_node(workspace: Workspace, grai_type: NodeID) -> NodeModel:
+    if grai_type.id is not None:
+        return NodeModel(**grai_type.__dict__)
+
+    return NodeModel.objects.filter(workspace=workspace).get(
+        namespace=grai_type.namespace, name=grai_type.name
+    )
+
+
+def deactivate(items: List[T]) -> List[T]:
+    updated = [item.update({"spec": {"is_active": False}}) for item in items]
```

Comment:
Based on the typing, if T is a Node/Edge (i.e. a subclass of pydantic BaseModel) this should be an error possibly being hidden by the `# type: ignore` flag on the return.

BaseModels don't have an update method and I believe this will replace all metadata in the spec with the single is_active value as well.

This also looks suspiciously like code I wrote for the client which, upon checking, is my fuck up :). I've created an issue ticket in linear to fix in the client but you can do this correctly here with

```
def deactivate(items: List[T]) -> List[T]:
    for item in items:
        item.spec.is_active = False
    return items
```

Reply:
Indeed I had just copied this from the client, corrected now.